### DISCUSSION TITLE
matplotlib ModuleNotFoundError

### DISCUSSION THREAD

**Discussion by c-hebert** [custom-components/pyscript#544] [Q&A]
Hello! I have been happily using pyscrips for two years, but After a year of letting my system live without intervention, I did a big update of my home assistant on a Raspberry Pi 2 weeks ago.
Everything works fine, apart from one annoying issue: I am using Python scripts (hacs/pyscript) to generate temperature plots that are then pushed on a distant server to be accessible from everywhere without opening remote access to my HA server.  and it doesn't work anymore

The log complains:  import matplotlib ^ ModuleNotFoundError: No module named 'matplotlib'
importing matplotlib. 

I tried adding this to config.yaml but without success:

pyscript:
  allow_all_imports: true
  hass_is_global: true

The very strange thing is that I can import matplotlib from within jupyterlab, so it is *somewhere*

**Comment by craigbarratt** [comment]
`pyscript` import doesn't work correctly on all modules - I'm not sure why.

As a workaround, can you put the `import matplotlib`, and the code that calls it, into a native (complied) python module, and then import that?  See the [docs](https://hacs-pyscript.readthedocs.io/en/latest/reference.html#importing).
